### PR TITLE
add `check_step_return`

### PR DESCRIPTION
**Pull request by Kallinteris-Andreas**
# Description

## Type of change
- [ ] Documentation only change (no code changed)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

**Comment by Kallinteris-Andreas** [comment]
I am confused, how is this similar with `env_step_passive_checker`???

- `env_step_passive_checker` :  checks that step returns a valid tuple `(obs, reward, terminated, truncated, info)`
- `check_step_return`: checks that the returns values deterministically 

The `RobotEnv`s fail as expect with `exact=True`, but not with `exact=False`
```sh
FAILED test.py::test_step[FetchSlide-v2] - AssertionError: step observation is not deterministic.
FAILED test.py::test_step[FetchPickAndPlace-v2] - AssertionError: step observation is not deterministic.
FAILED test.py::test_step[FetchReach-v2] - AssertionError: step observation is not deterministic.
FAILED test.py::test_step[FetchPush-v2] - AssertionError: step observation is not deterministic.
FAILED test.py::test_step[HandReach-v1] - AssertionError: step observation is not deterministic.
FAILED test.py::test_step[FetchSlideDense-v2] - AssertionError: step observation is not deterministic.
FAILED test.py::test_step[FetchPickAndPlaceDense-v2] - AssertionError: step observation is not deterministic.
FAILED test.py::test_step[FetchReachDense-v2] - AssertionError: step observation is not deterministic.
FAILED test.py::test_step[FetchPushDense-v2] - AssertionError: step observation is not deterministic.
FAILED test.py::test_step[HandReachDense-v1] - AssertionError: step observation is not deterministic.
```

**Comment by Kallinteris-Andreas** [comment]
Update: it now handles `nondeterministic` environments, by simply ignoring it the test

and runs `check_step_return` after `env_step_passive_checker`

File: gymnasium/utils/env_checker.py [hunk 1]
```diff
@@ -165,6 +165,39 @@ def check_reset_options(env: gym.Env):
         )
 
 
+def check_step_return(env: gym.Env, seed=123):
+    """Check that the environment steps dermistically after reset.
```
Comment (pseudo-rnd-thoughts) [inline gymnasium/utils/env_checker.py:169]: deterministic

File: gymnasium/utils/env_checker.py [hunk 2]
```diff
@@ -165,6 +165,39 @@ def check_reset_options(env: gym.Env):
         )
 
 
+def check_step_return(env: gym.Env, seed=123):
```
Comment (pseudo-rnd-thoughts) [inline gymnasium/utils/env_checker.py:168]: Could we change the name to `check_step_determinism`? We might want to change `check_reset_seed` to `check_reset_seed_determinism` to help make that clear as well

File: gymnasium/utils/env_checker.py [hunk 3]
```diff
@@ -165,6 +165,39 @@ def check_reset_options(env: gym.Env):
...
+    """Check that the environment steps dermistically after reset.
+
+    Note: This check assumes that seeded `reset()` is derministic (it must have passed `check_reset_seed`) and that `step()` returns valid values (passed `env_step_passive_checker`).
+    Note: A single step should be enough to assert that the state transition function is deterministic (at least for most environments).
+
+    Author: @Kallinteris-Andreas
```
Comment (pseudo-rnd-thoughts) [inline gymnasium/utils/env_checker.py:174]: Could you remove, most of the project doesn't include authorship links and if people want to know who wrote stuff can use git blame

File: gymnasium/utils/env_checker.py [hunk 4]
```diff
@@ -165,6 +165,39 @@ def check_reset_options(env: gym.Env):
...
+        AssertionError: The environment cannot be step determistially after resetting with a random seed,
+            or it truncates after 1 step.
+    """
+    if env.spec is not None and env.spec.nondeterministic is True:
+        return
+
+    env.action_space.seed(seed)
+    action = env.action_space.sample()
+
+    env.reset(seed=123)
```
Comment (pseudo-rnd-thoughts) [inline gymnasium/utils/env_checker.py:186]: Do we want to test with a range of seeds? Further, in `check_reset_seed` we test that the random number generators are equivalent

**Review by pseudo-rnd-thoughts — CHANGES_REQUESTED** [review]

File: gymnasium/utils/env_checker.py (hunk 4 above)
Comment (Kallinteris-Andreas) [reply gymnasium/utils/env_checker.py]: 1) I added an `assert` for the `_np_random`s being identical
2) No, I do not believe we should check a variety of checks (which would mean running the `check` functions with multiple seeds), `check_env` should be a "sanity check" for the API, not a comprehensive test, the responsibility of testing may cases (including edge cases) falls on the creator of the environment

**Review by pseudo-rnd-thoughts — CHANGES_REQUESTED** [review]
Looks good, could you add a test then good to merge.

**Comment by pseudo-rnd-thoughts** [comment]
@Kallinteris-Andreas Can you add the test then I'll merge, thanks

File: tests/utils/test_env_checker.py [hunk 5]
```diff
@@ -239,6 +239,41 @@ def test_check_reset_options():
...
+        [
+            UserWarning,
+            _reset_default_seed,
+            "The default seed argument in reset should be `None`, otherwise the environment will by default always be deterministic. Actual default: Error",
+        ],
+    ],
+)
+def test_check_step_determinism(test, func, message: str):
+    """Tests the check_step_determinism function."""
+    check_reset_options(GenericTestEnv(step_func=func))
```
Comment (pseudo-rnd-thoughts) [inline tests/utils/test_env_checker.py:274]: This runs `check_reset_options` not `check_step_seed_determinism`

**Review by pseudo-rnd-thoughts — CHANGES_REQUESTED** [review]

**Review by pseudo-rnd-thoughts — APPROVED** [review]
